src/tree/box.js

```diff
@@ -1,17 +1,64 @@
 import card from '../card/card';
 import selections from '../utils/selections';
+import { haveNoChildren } from '../utils/tree-utils';
 
-export default function box(divBox, o, appendNodes, cardStyling, selectionsAPI) {
+export const getSign = (d, { topId, isExpanded, expandedChildren }) => {
+  if (
+    (d.data.id === topId && isExpanded) ||
+    (d.parent && d.parent.data.id === topId && expandedChildren.includes(d.data.id))
+  ) {
+    return '-';
+  }
+
+  return '+';
+};
+
+export const getNewState = (d, { topId, isExpanded, expandedChildren }) => {
+  if (d.data.id === topId) {
+    // top
+    isExpanded = !isExpanded;
+    expandedChildren = [];
+  } else if (d.parent.data.id === topId) {
+    // children
+    if (expandedChildren.includes(d.data.id)) {
+      // Collapse if already exists in expandedChildren
+      expandedChildren.splice(expandedChildren.indexOf(d.data.id), 1);
+    } else if (d.parent.children.every(n => haveNoChildren(n.children))) {
+      // Add this node as expanded if possible
+      expandedChildren.push(d.data.id);
```

Comment:
this is incorrect, it should be:
if all expanded children has no children && d.children has no children
I don't think I got it correct when trying, as you have to fetch the nodes using the expandedChildren IDs.
This isn't a dealbreaker for now, but should be added to the backlog.

Reply:
Ah ok. That was not what I was going for though. This is just if all siblings expand vertically.
But as long as no siblings expand horizontally we can add them. I'll merge this and Iook at it int the next PR